IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -712,18 +809,24 @@ - (BOOL)removeItemsWithKey:(MSIDCacheKey *)key
     return [self removeStorageItemForKey:key context:context error:error];
 }
 
-- (NSDictionary *)accountAttributeForKey:(MSIDCacheKey *)key
+- (NSDictionary *)primaryAttributesForKey:(MSIDCacheKey *)key context:(id<MSIDRequestContext>)context error:(NSError **)error
 {
+    NSMutableDictionary *attributes = [NSMutableDictionary dictionary];
+    
     if (key.isShared)
     {
-        // Secret shareable item attribute: <keychainGroup>
-        return @{ (id)(kSecAttrAccount): self.keychainGroup};
+        // Shareable item attributes: <keychainGroup>
+        [attributes setObject:self.keychainGroup forKey:(id)kSecAttrAccount];
+        [attributes setObject:[self createAccess:key context:context error:error] forKey:(id)kSecAttrAccess];
```

Comment:
can createAccess:context:error: ever return nil?

Reply:
Yes it can. Added a check for this condition to avoid the crash.